tests/Assertion/ArraySubsetAssertionTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+declare(strict_types=1);
+
+namespace Zenstruck\Assert\Tests\Assertion;
+
+use PHPUnit\Framework\TestCase;
+use Zenstruck\Assert;
+use Zenstruck\Assert\Assertion\ArraySubsetAssertion;
+use Zenstruck\Assert\AssertionFailed;
+
+class ArraySubsetAssertionTest extends TestCase
+{
+    /** @dataProvider arraySubsetProvider */
+    public function testItAssertsArraySubset(array $needle, array $haystack): void
+    {
+        Assert::try(
+            static function () use ($needle, $haystack) {
+                (new ArraySubsetAssertion($needle, $haystack))();
+            }
+        );
+    }
+
+    public function arraySubsetProvider(): iterable
+    {
+        yield 'empty contains empty' => [[], []];
+        yield 'any array contains empty array' => [[], ['foo' => 'bar', 'bar' => 'foo']];
+        yield 'simple match' => [['foo' => 'bar'], ['foo' => 'bar', 'bar' => 'foo']];
+        yield 'even unordered keys is a subset' => [['bar' => 'foo', 'foo' => 'bar'], ['foo' => 'bar', 'bar' => 'foo']];
+        yield 'subset can be a list' => [[1, 2], [1, 2, 3]];
+        yield 'subset can be a deep list' => [[['foo' => 0], ['bar' => 1]], [['foo' => 0, 'bar' => 0], ['foo' => 1, 'bar' => 1], ['foo' => 2, 'bar' => 2]]];
+        yield 'subset can be a list with different keys' => [[1, 3], [1, 2, 3]];
+        yield 'deep match' => [
+            ['foo' => ['foo2' => ['foo3' => 'bar', 'list' => [1]]]],
+            ['foo' => ['foo2' => ['foo3' => 'bar', 'bar' => 'bar', 'list' => [1, 2, 3]], 'bar' => 'bar'], 'bar' => 'foo'],
+        ];
+    }
+
+    /** @dataProvider arrayNotSubsetProvider */
+    public function testItAssertsNotArraySubset(array $needle, array $haystack): void
+    {
+        try {
+            (new ArraySubsetAssertion($needle, $haystack))();
+        } catch (AssertionFailed $e) {
+            Assert::pass();
+
+            return;
+        }
+
+        Assert::fail('nop');
```

Comment:
This should be equivalent.

```suggestion
        Assert::not(new ArraySubsetAssertion($needle, $haystack));
```